hopp/tools/design/wind/power_curve_tools.py

```diff
@@ -0,0 +1,161 @@
+    return cp.tolist()
+
+def calculate_power_from_cp(wind_speed, cp_curve, rotor_diameter, rated_power_kW, air_density = 1.225):
+    """Calculate power curve from power coefficient curve (Cp).
+
+    Args:
+        wind_speed (list | np.ndarray): list of wind speeds in m/s
+        cp_curve (list | np.ndarray): power curve coefficients (Cp) at each wind speed in ``wind_speed``
+        rotor_diameter (float): rotor diameter of the turbine in meters.
+        air_density (float, Optional): Air density assumed for power-curve calculations in kg/m3.
+             Defaults to 1.225.
+    
+    Raises:
+        ValueError: if ``wind_speed`` and ``cp_curve`` are different lengths.
+
+    Returns:
+        list: turbine power (in kW) at each wind speed in ``wind_speed``
+    """
+
+    if len(wind_speed) != len(cp_curve):
+        raise ValueError("The length of the wind speed and coefficient of power vectors must be the same")
+
+    rotor_area = np.pi*((rotor_diameter/2)**2)
+    if isinstance(wind_speed, list):
+        wind_speed = np.array(wind_speed)
+    if isinstance(cp_curve, list):
+        cp_curve = np.array(cp_curve)
+    
+    # power available in the wind (kW)
+    p_wind = 0.5*air_density*rotor_area*(wind_speed**3)/1e3
+    power_kW = cp_curve*p_wind
+    power_kW = np.where(power_kW > rated_power_kW, rated_power_kW, power_kW)
+    power_kW = np.where(power_kW < 0, 0, power_kW)
+
+    return power_kW.tolist()
+
+def estimate_thrust_coefficient(wind_speed, cp_curve, plot=False, print_output=False):
+    """Calculate thrust coefficient curve (Ct) from power coefficient curve (Cp).
+
+    Args:
+        wind_speed (list | np.ndarray): list of wind speeds in m/s
+        cp_curve (list | np.ndarray): power curve coefficients (Cp) at each wind speed in ``wind_speed``
+        plot (bool, Optional): whether to plot Cp and Ct curve. Defaults to False.
+        print_output (bool, Optional): Whether to print Cp and Ct curves. Defaults to False.
+    
+    Raises:
+        ValueError: if ``wind_speed`` and ``cp_curve`` are different lengths.
+
+    Returns:
+        list: thrust curve coefficients (Ct) at each wind speed in ``wind_speed``
+    """
+    
+    # Check that the wind speed and the coefficient of power are the same length
+    if len(wind_speed) != len(cp_curve):
+        raise ValueError("The length of the wind speed and coefficient of power vectors must be the same")
+    N_wind = len(wind_speed)
+    ct_curve = list(np.zeros(N_wind))
+    
+    for i in range(N_wind):
+        # calculate induction factor a
+        # solve C_P = 4 * a * (1-a)**a  -> 4 * a**3 - 8 * a**2 + 4 * a - C_P = 0
+        roots = np.roots([4, -8, 4, -cp_curve[i]])
+
+        # Take root that is in range of a -> [0, 0.5]
+        a = roots[np.where(np.logical_and(roots>= 0, roots<= 0.5))]
+
+        # Calculate C_T = 4 * a * (1-a)
+        ct_curve[i] = np.round(4 * a * (1-a), 4)
+    
+    ct_flat = [ct[0] for ct in ct_curve]
```

Comment:
Could you update this to `ct_curve.flatten().tolist()`? The original comment is disconnected?